inference/embed-to-db-similar.py

```diff
@@ -0,0 +1,192 @@
+#     sims = [(torch.nn.functional.cosine_similarity(
+#                  out.last_hidden_state[0][0],
+#                  torch.tensor(state.loc[i,"mean"]),
+#                  dim=-1).item(),
+#              state.loc[i,"incident_id"]) for i in range(len(state))]
+#     return sims
+#
+# def inputted(whole_text, best_of=BEST_OF):
+#     sims = [j for j in sorted(test(whole_text), reverse=True)]
+#     if (best_of >= 0):
+#         return sims[:best_of]
+#     else:
+#         return sims
+#
+# # What to do to correctly formatted input event_text
+# def process(event_text, best_of=BEST_OF):
+#     # return tokenizer(event_text)
+#     return inputted(event_text, best_of)
+
+
+# Define lambda handler
+def handler(event, context):
+    # Starting point for response formatting
+    result = {
+        "isBase64Encoded": False,
+        "statusCode": 500,
+        "headers": {"Content-Type": "application/json"},
+        "multiValueHeaders": {},
+        "body": {"warnings": []}
+    }
+
+    # Get input from body or query string
+    try:
+        if ('embed' in event):
+            embed_text = event['embed']
+        elif ('body' in event and event['body'] != '' and 'embed' in json.loads(event['body'])):
+            embed_text = json.loads(event['body'])['embed']
+        elif ('queryStringParameters' in event and 'embed' in event['queryStringParameters']):
+            embed_text = event['queryStringParameters']['embed']
+        else:
+            result['statusCode'] = 500
+            result['body'] = {'msg': 'Error! Valid input text not provided!'}
+            result['headers']['Content-Type'] = "application/json"
+            return json.dumps(result)
+    except:
+        result['statusCode'] = 500
+        result['body'] = {'msg': 'Error! Parsing input gave exception!'}
+        result['headers']['Content-Type'] = "application/json"
+        return json.dumps(result)
+
+    # Get "best of" value from body or query string (or <0 for full list)
+    best_of = BEST_OF
+    if ('num' in event):
+        best_of = event['num']
+    elif ('body' in event and event['body'] != '' and 'num' in json.loads(event['body'])):
+        best_of = json.loads(event['body'])['num']
+    elif ('queryStringParameters' in event and 'num' in event['queryStringParameters']):
+        best_of = event['queryStringParameters']['num']
+
+    # Assign to best of if possible
+    try:
+        if (best_of != BEST_OF):  # if input found (type/value mismatch)
+            best_of = int(best_of)
+    except ValueError:
+        best_of = BEST_OF
+        result['body']['warnings'].append(
+            f'Provided value for "num" invalid, using default of {BEST_OF}.')
+    if (best_of == 0):
+        result['body']['warnings'].append(
+            f'Zero results requested with the "num" value of 0. Use value <0 for maximum possible.')
+
+    # Handle unicode in event_text and parse it to a list (removing up to 10 levels of nested quotes, just in case)
```

Comment:
Here it decodes the embed_text string from the body of the request into a literal list by using literal_eval, but it must do this a couple times because AWS passing around can sometimes add a couple layers of quotes around the string